pkg/provider/data_source_cluster.go

```diff
@@ -237,16 +242,20 @@ func (c *ClusterData) Read(ctx context.Context, d *schema.ResourceData, meta any
 
 	clusterName, ok := d.Get("cluster_name").(string)
 	if !ok {
-		return diag.FromErr(errors.New("Unable to find cluster name"))
+		return diag.FromErr(errors.New("unable to find cluster name"))
+	}
+	projectId, ok := d.Get("project_id").(string)
+	if !ok {
+		return diag.FromErr(errors.New("unable to find a project ID"))
```

Comment:
> unable to find a project ID

just a nit, but it might be more accurate to say 
> unable to find project with id '%s'

or something?

Reply:
This actually checks the `project_id` value in the configuration. So, it doesn't check if a project with this `project_id` exists or not. 

I'm thinking about checking if the `project_id` project is reachable or not, even before the calls. I will check tomorrow.

Reply:
This check is actually pretty useless. `project_id` is always a required field, It should always be provided by the user to run the terraform command on that resource or data-source. Only after that, the validation function will run and validate that the project_id starts with `prj_`, followed by 16 letters. So, this error check only shows if the 
 `d.Get("project_id").(string)` is failed. I think this is enough to discard this `if !ok {` check. 

We are already doing it in the region resource btw, for example https://github.com/EnterpriseDB/terraform-provider-biganimal/blob/main/pkg/provider/resource_region.go#L104-L106

I think it's fair to remove them. What do you think, @nstott?